### PR TITLE
Revert "Support split broadcast join condition into ast and non-ast […

### PR DESCRIPTION
**Pull request by winningsix**
…databricks] (#9760)"

This reverts commit 7c307d489b099bd1d73f8ec4b9b953a05ca24dec.

There is an issue reported by NDS https://github.com/NVIDIA/spark-rapids/issues/10047 WIP on a fix.

### CODE REVIEW

**Comment by winningsix** [comment]
build

**Comment by winningsix** [comment]
build

File: tests/src/test/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala [hunk 1]
```diff
@@ -66,7 +66,7 @@ class MockTaskContext(taskAttemptId: Long, partitionId: Int) extends TaskContext
 
   override private[spark] def killTaskIfInterrupted(): Unit = {}
 
-  override private[spark] def getKillReason() = None
+  override def getKillReason() = None
```
Comment (jbrennan333) [inline tests/src/test/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala:69]: I don't think this change was part of the original commit, was it?

**Review by jbrennan333 — COMMENTED** [review]
Looks like there might be an extra change here?

File: tests/src/test/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala (hunk 1 above)
Comment (winningsix) [reply tests/src/test/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala:69]: Yes, this is to fix DBX build.

**Comment by winningsix** [comment]
build

File: tests/src/test/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala (hunk 1 above)
Comment (winningsix) [reply tests/src/test/scala/org/apache/spark/sql/rapids/metrics/source/MockTaskContext.scala:69]: It seems going to be fixed in https://github.com/NVIDIA/spark-rapids/pull/10070 Will update accordingly once merged.

**Review by jbrennan333 — APPROVED** [review]

**Pull request merged**